backend/src/services/mirroredModel.ts

```diff
@@ -77,6 +86,67 @@ export async function exportModel(
   log.debug({ modelId, semvers }, 'Successfully finalized zip file.')
 }
 
+export async function importModel(_user: UserInterface, mirroredModelId: string, payloadUrl: string) {
+  if (mirroredModelId === '') {
+    throw BadReq('Missing mirrored model ID.')
+  }
+  let sourceModelId
+
+  let res: Response
+  try {
+    res = await fetch(payloadUrl)
+  } catch (err) {
+    throw InternalError('Unable to get the file.', { err })
+  }
+  if (!res.ok) {
+    throw InternalError('Unable to get zip file.', { response: { status: res.status, body: await res.text() } })
+  }
+
+  if (!res.body) {
+    throw InternalError('Unable to get the file.')
+  }
+
+  const modelCards: ModelCardRevisionInterface[] = []
+  const test = new Uint8Array(await res.arrayBuffer())
+  const zipContent = fflate.unzipSync(test, {
+    filter(file) {
+      return /[0-9]+.json/.test(file.name)
+    },
+  })
+  Object.keys(zipContent).forEach(function (key) {
+    const { modelCard, sourceModelId: newSourceModelId } = parseModelCard(
+      Buffer.from(zipContent[key]).toString('utf8'),
+      mirroredModelId,
+      sourceModelId,
+    )
+    if (!sourceModelId) {
+      sourceModelId = newSourceModelId
+    }
+    modelCards.push(modelCard)
+  })
+
+  await Promise.all(modelCards.map((card) => saveImportedModelCard(card, sourceModelId)))
+  await setLatestImportedModelCard(mirroredModelId)
+  return { mirroredModelId, sourceModelId, modelCardVersions: modelCards.map((modelCard) => modelCard.version) }
+}
+
+function parseModelCard(modelCardString: string, mirroredModelId: string, sourceModelId?: string) {
```

Comment:
For readability it might be worth clarifying that `modelCardString` is `modelCardJson`. At a glance "string" could refer to a number of things related to the model card.